openmcyclus/depletion.py

```diff
@@ -3,156 +3,192 @@
+                assembly_number = child.attrib['name'][underscore_index + 1:]
+                material_ids.append(child.attrib['id'])
+                if child.attrib['name'][:underscore_index] == 'assembly':
+                    for material in child.findall('nuclide'):
+                        child.remove(material)
+                    new_comp = comp_list[int(assembly_number) - 1]
+                    for nuclide in new_comp:
+                        Z = math.floor(nuclide / int(1e7))
+                        A = math.floor((nuclide - Z * int(1e7)) / int(1e4))
+                        m = nuclide - Z * int(1e7) - A * int(1e4)
+                        nucname = openmc.data.gnd_name(Z, A, m)
+                        new_nuclide = f"""<nuclide wo="{str(new_comp[nuclide]*100)}" name="{nucname}" />"""
+                        new_nuclide_xml = ET.fromstring(new_nuclide)
+                        child.insert(1, new_nuclide_xml)
+        ET.indent(openmc_root)
+        openmc_materials.write(str(path + "materials.xml"))
+        return material_ids
+
+    def read_microxs(self, path):
         '''
         Reads .csv file with microscopic cross sections. The
         csv file is assumed to be named "micro_xs.csv". This will need to
         be relative to Cyclus input file location
 
         Parameters:
         -----------
+        path: str
+            path of directory holding the files for/from OpenMC        
 
         Returns:
         --------
         microxs: object
             microscopic cross section data
         '''
-        microxs = od.MicroXS.from_csv(self.path / "micro_xs.csv")
+        microxs = od.MicroXS.from_csv(str(path + "micro_xs.csv"))
         return microxs
 
-    def run_depletion(self):
+    def run_depletion(self, path):
         '''
         Run the IndependentOperator class in OpenMC to perform
         transport-independent depletion.
 
         Parameters:
-        -----------
+        -----------        
+        path: str
+            path of directory holding the files for/from OpenMC
 
         Outputs:
         --------
         depletion_results.h5: database
             HDF5 data base with the results of the depletion simulation
         '''
-        model = self.read_model()
-        micro_xs = self.read_microxs()
+        model = self.read_model(path)
+        micro_xs = self.read_microxs(path)
         ind_op = od.IndependentOperator(model.materials, micro_xs,
-                                        str(self.path / self.chain_file))
-        ind_op.output_dir = self.path
-        integrator = od.PredictorIntegrator(ind_op, np.ones(
-            self.timesteps*30), power=self.power*1000*self.conversion_factor, 
+                                        str(path + self.chain_file))
+        ind_op.output_dir = path
+        integrator = od.PredictorIntegrator(
+            ind_op,
+            np.ones(
+                self.timesteps *
+                30),
```

Comment:
where does 30 come from? Does this convert months to days (approximately)?

Reply:
Yes. The `self.timesteps` is the number of months in a cycle, but OpenMC doesn't accept month as a timestep unit. I will be changing this (in a commit for this PR) to `np.ones(self.timesteps)*30, so that each time step for OpenMC is 30 days, instead of using 1 day timesteps and having to multiply the number of time steps by 30 and significantly increase the run time.